pkg/lvm/lvm2app.go

```diff
@@ -892,23 +574,112 @@ func ListVolumeGroupNames() ([]string, error) {
+	result := new(vgsOutput)
+	if err := run("vgs", result, "--options=vg_uuid"); err != nil {
+		return nil, err
+	}
+	var uuids []string
+	for _, report := range result.Report {
+		for _, vg := range report.Vg {
+			uuids = append(uuids, vg.UUID)
+		}
+	}
+	return uuids, nil
 }
 
 // CreatePhysicalVolume creates a physical volume of the given device.
 func CreatePhysicalVolume(dev string) (*PhysicalVolume, error) {
-	return defaultHandle.CreatePhysicalVolume(dev)
+	if err := run("pvcreate", nil, dev); err != nil {
+		return nil, fmt.Errorf("lvm: CreatePhysicalVolume: %v", err)
+	}
+	return &PhysicalVolume{dev}, nil
+}
+
+type pvsOutput struct {
+	Report []struct {
+		Pv []struct {
+			Name   string `json:"pv_name"`
+			VgName string `json:"vg_name"`
+		} `json:"pv"`
+	} `json:"report"`
 }
 
 // ListPhysicalVolumes lists all physical volumes.
 func ListPhysicalVolumes() ([]*PhysicalVolume, error) {
-	return defaultHandle.ListPhysicalVolumes()
+	result := new(pvsOutput)
+	if err := run("pvs", result); err != nil {
+		return nil, err
+	}
+	var pvs []*PhysicalVolume
+	for _, report := range result.Report {
+		for _, pv := range report.Pv {
+			pvs = append(pvs, &PhysicalVolume{pv.Name})
+		}
+	}
+	return pvs, nil
 }
 
 // LookupPhysicalVolume returns a physical volume with the given name.
 func LookupPhysicalVolume(name string) (*PhysicalVolume, error) {
-	return defaultHandle.LookupPhysicalVolume(name)
+	result := new(pvsOutput)
+	if err := run("pvs", result, "--options=pv_name", name); err != nil {
+		if IsPhysicalVolumeNotFound(err) {
+			return nil, ErrPhysicalVolumeNotFound
+		}
+		return nil, err
+	}
+	for _, report := range result.Report {
+		for _, pv := range report.Pv {
+			return &PhysicalVolume{pv.Name}, nil
+		}
+	}
+	return nil, ErrPhysicalVolumeNotFound
 }
 
 // Extent sizing for linear logical volumes:
 // https://github.com/Jajcus/lvm2/blob/266d6564d7a72fcff5b25367b7a95424ccf8089e/lib/metadata/metadata.c#L983
+
+func run(cmd string, v interface{}, extraArgs ...string) error {
+	var args []string
+	if v != nil {
+		args = append(args, "--reportformat=json")
+		args = append(args, "--units=b")
+		args = append(args, "--nosuffix")
+	}
+	args = append(args, extraArgs...)
+	c := exec.Command(cmd, args...)
+	log.Printf("Executing: %v", c)
+	stdout, stderr := new(bytes.Buffer), new(bytes.Buffer)
+	c.Stdout = stdout
+	c.Stderr = stderr
+	if err := c.Run(); err != nil {
+		errstr := ignoreWarnings(stderr.String())
+		log.Print("stdout: " + stdout.String())
+		log.Print("stderr: " + errstr)
+		return errors.New(errstr)
+	}
+	stdoutbuf := stdout.Bytes()
+	stderrbuf := stderr.Bytes()
+	errstr := ignoreWarnings(string(stderrbuf))
+	log.Printf("stdout: " + string(stdoutbuf))
+	log.Printf("stderr: " + errstr)
+	if v != nil {
+		if err := json.Unmarshal(stdoutbuf, v); err != nil {
+			return fmt.Errorf("%v: [%v]", err, string(stdoutbuf))
+		}
+	}
+	return nil
+}
+
+func ignoreWarnings(str string) string {
```

Comment:
nit: this isn't very efficient. although it does the job, we may want to refactor this at some point

Reply:
My thinking w.r.t. performance is usually: compared to the overhead of serving an HTTP request or executing a cli command string manipulation is typically free.